gtsfm/loader/hilti_loader.py

```diff
@@ -257,35 +400,38 @@ def get_relative_pose_prior(self, i1: int, i2: int) -> Optional[PosePrior]:
 
-        # TODO(Frank): this should come from constraints?
-
         if rig_idx_for_i1 == rig_idx_for_i2:
             i1_T_imu: Pose3 = self._cam_T_imu_poses[cam_idx_for_i1]
             i2_T_imu: Pose3 = self._cam_T_imu_poses[cam_idx_for_i2]
-            i2Ti1 = i2_T_imu.inverse().between(i1_T_imu.inverse())
-            # TODO: add covariance
-            return PosePrior(value=i2Ti1, covariance=HARD_RELATIVE_POSE_PRIOR_SIGMA, type=PosePriorType.HARD_CONSTRAINT)
-        elif rig_idx_for_i1 in self._w_T_imu and rig_idx_for_i2 in self._w_T_imu:
-            w_T_i1 = self._w_T_imu[rig_idx_for_i1] * self._cam_T_imu_poses[cam_idx_for_i1].inverse()
-            w_T_i2 = self._w_T_imu[rig_idx_for_i2] * self._cam_T_imu_poses[cam_idx_for_i2].inverse()
-            i2Ti1 = w_T_i2.between(w_T_i1)
-            # TODO: add covariance
-            return PosePrior(value=i2Ti1, covariance=SOFT_RELATIVE_POSE_PRIOR_SIGMA, type=PosePriorType.SOFT_CONSTRAINT)
-
-        return None
+            i1Ti2 = i1_T_imu * i2_T_imu.inverse()
+            return PosePrior(value=i1Ti2, covariance=HARD_POSE_PRIOR_COVARIANCE, type=PosePriorType.HARD_CONSTRAINT)
+        elif cam_idx_for_i1 == 2 and cam_idx_for_i2 == 2 and (rig_idx_for_i1, rig_idx_for_i2) in self._constraints:
+            constraint = self._constraints[(rig_idx_for_i1, rig_idx_for_i2)]
+            imui1_T_imui2 = constraint.aTb
+            i1Ti2 = self._cam_T_imu_poses[2] * imui1_T_imui2 * self._cam_T_imu_poses[2].inverse()
+            cov_i1Ti2 = self._cam2_J_imu_relative @ constraint.cov @ self._cam2_J_imu_relative.T
+
+            # Rig is stationary, add a hard constraint.
+            if i1Ti2.equals(Pose3(), 1e-6):
+                return PosePrior(value=i1Ti2, covariance=cov_i1Ti2, type=PosePriorType.HARD_CONSTRAINT)
+
+            return PosePrior(value=i1Ti2, covariance=cov_i1Ti2, type=PosePriorType.SOFT_CONSTRAINT)
+        elif cam_idx_for_i1 == 2 and cam_idx_for_i2 == 2:
+            # CAM2-CAM2 backbone as soft prior
+            w_T_imui1 = self._fastlio_poses[rig_idx_for_i1]
+            w_T_imui2 = self._fastlio_poses[rig_idx_for_i2]
+            imui1_T_imui2 = w_T_imui1.between(w_T_imui2)
+            i1Ti2 = self._cam_T_imu_poses[2] * imui1_T_imui2 * self._cam_T_imu_poses[2].inverse()
 
-    def get_absolute_pose_prior(self, idx: int) -> Optional[PosePrior]:
-        rig_idx: int = self.rig_from_image(idx)
-        cam_idx: int = self.camera_from_image(idx)
-
-        if rig_idx in self._w_T_imu:
-            w_T_cam = self._w_T_imu[rig_idx] * self._cam_T_imu_poses[cam_idx].inverse()
             return PosePrior(
-                value=w_T_cam, covariance=SOFT_ABSOLUTE_POSE_PRIOR_SIGMA, type=PosePriorType.SOFT_CONSTRAINT
+                value=i1Ti2, covariance=CAM2_BACKBONE_POSE_PRIOR_COVARIANCE, type=PosePriorType.SOFT_CONSTRAINT
             )
 
         return None
 
+    def get_absolute_pose_prior(self, idx: int) -> Optional[PosePrior]:
```

Comment:
should we remove absolute pose priors everywhere? are they used anywhere? can be done in a separate PR.

Reply:
Not using it anywhere for now. @dellaert what do you think?

Reply:
lets remove it if we are not using it. I remember Frank was also not keen on having it.

Reply:
We use it in 1dsfm.